Sources/HStack.swift

```diff
@@ -41,6 +60,11 @@ open class HStack: Stack {
                 stackableWidth = fixedSizeStackable.size.width
             } else if let fixedSizeStack = stackable as? Stack, let stackWidth = fixedSizeStack.width {
                 stackableWidth = stackWidth
+            } else if let item = stackable as? StackableItem, distribution == .fillUsingIntrinsicContentSize {
+                stackableWidth = min(
```

Comment:
This change enables us to have the items size themselves to their intrinsic content size width.  If however based on previous elements and width available it is less than the intrinsic content size width we will go with the space available.

This works in our favour as the height itself is calculate further down based on width so for labels for example they can wrap multiple lines not just be forced to truncate etc.